### PR TITLE
Add 'updateFromDF()' to magmaR

### PR DESCRIPTION
**Pull request by dtm2451**
Adds a new convenience function to magmaR that essentially mimics `etna administrate models attributes update_from_csv`, just in R.

I put it together after discovering a wonderfully relevant chunk of code from a past `magmaR::update()` upload.

Not the highest priority to finish, unless a user requests such functionality.

To Do:
- [x] Add to the Upload vignette
- [x] Unit tests

### CODE REVIEW

File: etna/packages/magmaR/R/update.R [hunk 1]
```diff
@@ -106,7 +107,7 @@ updateValues <- function(
         num_new <- length(rec_names_new)
         
         ### Summarize for NEW records
-        cat("For model \"", modelName, "\", this update() will create ", num_new, " NEW records:\n    ",
+        cat("For model \"", modelName, "\", this update() will create (or update) ", num_new, " NEW (or orphan) records:\n    ",
```
Comment (dtm2451) [inline etna/packages/magmaR/R/update.R:110]: Addresses #431 by adjusting the 'NEW' records section of the update summary to report that they could also be orphan records.

**Comment by coleshaw** [comment]
Do you think this addresses the mystery issue with the extra `.1`, `.2`, etc. records being created?

Perhaps @graft is the most qualified, since he's worked with R more?

**Comment by graft** [comment]
> Do you think this addresses the mystery issue with the extra `.1`, `.2`, etc. records being created?
> 
> Perhaps @graft is the most qualified, since he's worked with R more?

I don't think this addresses those issues? Or at least I don't see how. But @dtm2451 closed #541 so maybe it does?

**Comment by dtm2451** [comment]
I think those were created from an improper input where that was the actual request.  That + plus Bushra and I must have ignored the Summary of the update that was about to be run, as there isn't much manipulation after the stage where records to be updated are summarized.  Certainly no name-setting where ".1"/".2" uniquefication would happen.

I've not found anywhere in the `updateValues()` function where something like that would have happened, so I'm relatively confident it was upstream of magmaR.  Sorry, as I could have commented it, but that's why I closed #541.

**Review by graft — APPROVED** [review]

**Comment by dtm2451** [comment]
:+1: Beginning the process of re-submitting to CRAN.  I'll wait to merge until the first round of checks are done.

**Comment by dtm2451** [comment]
Submitted to CRAN

**Pull request merged**